src/cider/log/specs.clj

```diff
@@ -0,0 +1,286 @@
+                   :cider.log.framework/levels
+                   :cider.log.framework/log-fn
+                   :cider.log.framework/name
+                   :cider.log.framework/remove-appender-fn
+                   :cider.log.framework/root-logger
+                   :cider.log.framework/website-url]))
+
+(s/def :cider.log.appender/id string?)
+(s/def :cider.log.appender/levels (s/coll-of :cider.log/level))
+(s/def :cider.log.appender/logger string?)
+(s/def :cider.log.appender/size pos-int?)
+(s/def :cider.log.appender/threshold (s/and nat-int? #(< % 100)))
+
+(s/def :cider.log/appender
+  (s/keys :req-un [:cider.log.appender/id]
+          :opt-un [:cider.log.appender/levels
+                   :cider.log.appender/logger
+                   :cider.log.appender/size
+                   :cider.log.appender/threshold]))
+
+(s/def :cider.log.consumer/callback ifn?)
+(s/def :cider.log.consumer/filter (s/map-of string? any?))
+(s/def :cider.log.consumer/id string?)
+
+(s/def :cider.log/consumer
+  (s/keys :req-un [:cider.log.consumer/id]
+          :opt-un [:cider.log.consumer/callback
+                   :cider.log.consumer/filter]))
+
+(s/def :cider.log.event/argument any?)
+(s/def :cider.log.event/arguments (s/coll-of :cider.log.event/argument :kind vector?))
+(s/def :cider.log.event/id uuid?)
+(s/def :cider.log.event/level simple-keyword?)
+(s/def :cider.log.event/logger string?)
+(s/def :cider.log.event/mdc (s/map-of string? string?))
+(s/def :cider.log.event/message string?)
+(s/def :cider.log.event/thread string?)
+(s/def :cider.log.event/timestamp pos-int?)
+
+(s/def :cider.log/event
+  (s/keys :req-un [:cider.log.event/arguments
+                   :cider.log.event/id
+                   :cider.log.event/level
+                   :cider.log.event/logger
+                   :cider.log.event/mdc
+                   :cider.log.event/message
+                   :cider.log.event/thread
+                   :cider.log.event/timestamp]))
+
+;; cider.log.framework
+
+(s/fdef framework/add-appender
+  :args (s/cat :framework :cider.log/framework :appender :cider.log/appender)
+  :ret :cider.log/framework)
+
+(s/fdef framework/add-consumer
+  :args (s/cat :framework :cider.log/framework :appender :cider.log/appender :consumer :cider.log/consumer)
+  :ret :cider.log/framework)
+
+(s/fdef framework/clear-appender
+  :args (s/cat :framework :cider.log/framework :appender :cider.log/appender)
+  :ret :cider.log/framework)
+
+(s/fdef framework/consumer
+  :args (s/cat :framework :cider.log/framework :appender :cider.log/appender :consumer :cider.log/consumer)
+  :ret (s/nilable :cider.log/consumer))
+
+(s/fdef framework/event
+  :args (s/cat :framework :cider.log/framework :appender :cider.log/appender :id :cider.log.event/id)
+  :ret (s/nilable :cider.log/event))
+
+(s/fdef framework/events
+  :args (s/cat :framework :cider.log/framework :appender :cider.log/appender)
+  :ret (s/coll-of :cider.log/event))
+
+(s/fdef framework/log
```

Comment:
The `framework/log` function isn't actually part of the hot path. It' mostly a convenience function that can be used in the REPL and in the tests to abstract the differences of the log frameworks. Since I mostly wanted those specs during development and testing I agree with moving them to the test dir.